api/v1beta1/cluster_types.go

```diff
@@ -40,6 +40,238 @@ const (
 	ClusterKind = "Cluster"
 )
 
+// Cluster's Available condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterAvailableV1Beta2Condition is true if the Cluster is not deleted, and RemoteConnectionProbe, InfrastructureReady,
+	// ControlPlaneAvailable, WorkersAvailable, TopologyReconciled (if present) conditions are true.
+	// If conditions are defined in spec.availabilityGates, those conditions must be true as well.
+	ClusterAvailableV1Beta2Condition = AvailableV1Beta2Condition
+
+	// ClusterAvailableInternalErrorV1Beta2Reason surfaces unexpected error when computing the Available condition.
+	ClusterAvailableInternalErrorV1Beta2Reason = InternalErrorV1Beta2Reason
+)
+
+// Cluster's TopologyReconciled condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterTopologyReconciledV1Beta2Condition is true if the topology controller is working properly.
+	// Note: This condition is added only if the Cluster is referencing a ClusterClass / defining a managed Topology.
+	ClusterTopologyReconciledV1Beta2Condition = "TopologyReconciled"
+)
+
+// Cluster's InfrastructureReady condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterInfrastructureReadyV1Beta2Condition mirrors Cluster's infrastructure Ready condition.
+	ClusterInfrastructureReadyV1Beta2Condition = InfrastructureReadyV1Beta2Condition
+
+	// ClusterInfrastructureInvalidConditionReportedV1Beta2Reason surfaces a infrastructure Ready condition (read from an infra cluster object) which is invalid
+	// (e.g. its status is missing).
+	ClusterInfrastructureInvalidConditionReportedV1Beta2Reason = InvalidConditionReportedV1Beta2Reason
+
+	// ClusterInfrastructureReadyNoReasonReportedV1Beta2Reason applies to a infrastructure Ready condition (read from an infra cluster object) that reports no reason.
+	ClusterInfrastructureReadyNoReasonReportedV1Beta2Reason = NoReasonReportedV1Beta2Reason
+
+	// ClusterInfrastructureInternalErrorV1Beta2Reason surfaces unexpected failures when reading an infra cluster object.
+	ClusterInfrastructureInternalErrorV1Beta2Reason = InternalErrorV1Beta2Reason
+
+	// ClusterInfrastructureDoesNotExistV1Beta2Reason surfaces when a referenced infrastructure object does not exist.
+	// Note: this could happen when creating the Cluster. However, this state should be treated as an error if it lasts indefinitely.
+	ClusterInfrastructureDoesNotExistV1Beta2Reason = ObjectDoesNotExistV1Beta2Reason
+
+	// ClusterInfrastructureDeletedV1Beta2Reason surfaces when a referenced infrastructure object has been deleted.
+	// Note: controllers can't identify if the infrastructure object was deleted by the controller itself, e.g.
+	// during the deletion workflow, or by a users.
+	ClusterInfrastructureDeletedV1Beta2Reason = ObjectDeletedV1Beta2Reason
```

Comment:
how is semantically different from the one above?

Reply:
We use "Deleted" when we know for certain that the object existed at some point, we use "DoesNotExists" when we don't know if it ever existed